### PR TITLE
Add retention leases related recovery tests

### PR DESCRIPTION
**Pull request by mkleen**
This adds backports of the two retention leases related recovery tests from `RecoveryIT.java`:

- test_retention_leases_established_when_promoting_primary

https://github.com/elastic/elasticsearch/blob/master/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java#L328

- test_retention_leases_established_when_relocating_primary

https://github.com/elastic/elasticsearch/blob/master/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java#L350

Note:
This only works with soft deletes disabled, we are missing some more recent backports.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by mkleen** [comment]
retest this please

File: tests/bwc/test_recovery.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class RecoveryTest(NodeProvider, unittest.TestCase):
 
     NUMBER_OF_NODES = 3
 
-    def assert_busy(self, assertion, timeout=120, f=2.0):
+    def assert_busy(self, assertion, timeout=180, f=2.0):
```
Comment (mkleen) [inline tests/bwc/test_recovery.py:23]: This is increase is not related to these tests, other tests are timing out.

**Comment by mkleen** [comment]
retest this please

**Review by seut — APPROVED** [review]
👍
